src/Form/SettingFormType.php

```diff
@@ -86,13 +86,21 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         'attr' => ['rows' => 12],
                     ]);
             } elseif ($model->getType()->equals(Type::CHOICE())) {
+                $choices = [];
+                foreach (array_flip($model->getChoices()) as $label => $value) {
```

Comment:
non associative array is when keys are continuous integer from` 0` to `count($array) - 1`.
In some cases maybe labels are meat to be integers.

I would upgrade this to something like:

```
$choices = array_values($model->getChoices()) === $model->getChoices()
    ? array_combine($model->getChoices(), $model->getChoices())
    : $model->getChoices();
```

`choices: { choice1: 'Choice 1', choice2: 'Choice 2'}` this should not be valid. Keys should be labels, and values - values :)

Reply:
Yes, you're right, code updated.